dtfs-central-api/src/v1/controllers/utilisation-report-service/get-selected-fee-records-details.controller/helpers.test.ts

```diff
@@ -0,0 +1,125 @@
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build())
+        .withFeesPaidToUkefForThePeriod(200)
+        .withFeesPaidToUkefForThePeriodCurrency('USD')
+        .withPaymentCurrency('USD')
+        .build();
+      // Act
+      const result = await mapToSelectedFeeRecordDetails('123', aReportPeriod(), [feeRecord]);
+
+      // Assert
+      expect(result.feeRecords[0].reportedPayment).toEqual<CurrencyAndAmount>({ amount: 200, currency: 'USD' });
+    });
+
+    it('sets the reported payment to the fees paid to ukef for the period converted into the payment currency when the payment currency does not match the fees paid to ukef for the period currency', async () => {
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build())
+        .withFeesPaidToUkefForThePeriod(2200)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .withPaymentCurrency('GBP')
+        .withPaymentExchangeRate(1.1)
+        .build();
+
+      // Act
+      const result = await mapToSelectedFeeRecordDetails('123', aReportPeriod(), [feeRecord]);
+
+      // Assert
+      expect(result.feeRecords[0].reportedPayment).toEqual<CurrencyAndAmount>({ amount: 2000, currency: 'GBP' });
+    });
+
+    it('maps the fees paid to ukef for the period to the reported fee', async () => {
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build())
+        .withFeesPaidToUkefForThePeriod(2200)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .build();
+      // Act
+      const result = await mapToSelectedFeeRecordDetails('123', aReportPeriod(), [feeRecord]);
+
+      // Assert
+      expect(result.feeRecords[0].reportedFee).toEqual<CurrencyAndAmount>({ amount: 2200, currency: 'EUR' });
+    });
+
+    it('calculates total reported payments in payment currency by adding up reported payments of all fee records', async () => {
+      // Arrange
+      const feeRecordWithDifferingCurrencies = FeeRecordEntityMockBuilder.forReport(
+        UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build(),
+      )
+        .withFeesPaidToUkefForThePeriod(2200)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .withPaymentCurrency('GBP')
+        .withPaymentExchangeRate(1.1)
+        .build();
+
+      const feeRecordWithMatchingCurrencies = FeeRecordEntityMockBuilder.forReport(
+        UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build(),
+      )
+        .withFeesPaidToUkefForThePeriod(1000)
+        .withFeesPaidToUkefForThePeriodCurrency('GBP')
+        .withPaymentCurrency('GBP')
+        .build();
```

Comment:
I've changed it, I can't decide if it matters since the method doesn't use that fact at all and shouldn't, but it does match the real use case data closer